pandas/core/indexes/datetimes.py

```diff
@@ -274,6 +282,35 @@ class DatetimeIndex(DatelikeOps, TimelikeOps, DatetimeIndexOpsMixin,
     TimedeltaIndex : Index of timedelta64 data
     PeriodIndex : Index of Period data
     pandas.to_datetime : Convert argument to datetime
+
+    Examples
+    --------
+    Generate a date range with month end frequency
+
+    >>> pd.DatetimeIndex(start='2018-3-10', freq='M', periods=4)
```

Comment:
All the example are for generating range, but I think the more typical use case for DatetimeIndex is to create an index from an array-like. Can you add examples for that as well?

Reply:
(and for creating ranges, I would actually just refer to `date_range`